src/frontend/src/lib/components/loaders/LoaderCollections.svelte

```diff
@@ -71,6 +73,51 @@
 		}
 	};
 
+	const handleErc1155 = async ({
+		contracts,
+		customTokens,
+		network,
+		identity
+	}: {
+		contracts: OwnedContract[];
+		customTokens: CustomToken[];
+		network: EthereumNetwork;
+		identity: Identity;
+	}) => {
+		const tokens = contracts.reduce<SaveErc1155CustomToken[]>((acc, contract) => {
+			const existingToken = customTokens.find(({ token }) => {
+				if (!('Erc1155' in token)) {
+					return false;
+				}
+
+				const {
+					Erc1155: { token_address: tokenAddress, chain_id: tokenChainId }
+				} = token;
+
+				return tokenAddress === contract.address && tokenChainId === network.chainId;
+			});
+			if (nonNullish(existingToken)) {
+				return acc;
+			}
+
+			const newToken: SaveErc1155CustomToken = {
+				address: contract.address,
+				network,
+				enabled: !contract.isSpam
+			};
+			acc.push(newToken);
```

Comment:
We usually do it like this:

```suggestion
			acc = [...acc, newToken];
```